src/esp/gfx/RenderTarget.cpp

```diff
@@ -128,14 +128,46 @@ struct RenderTarget::Impl {
     depthUnprojectionMesh_.draw(*depthShader_);
   }
 
+#ifdef ESP_BUILD_WITH_TRIANGLE_SENSOR
+  void initTriangleBuffer() {
+    if (triangleMesh_.id() == 0) {
+      // triangleIdBuffer_ = Mn::GL::Renderbuffer{};
+      // triangleIdBuffer_.setStorage(Mn::GL::RenderbufferFormat::R32I,
+      //                             framebufferSize());
+
+      // triangleFrameBuffer_ =
+      //    Mn::GL::Framebuffer{{{}, framebufferSize()}};
+      // triangleFrameBuffer_
+      //     .attachRenderbuffer(TriangleIdBuffer, triangleIdBuffer_)
+      //     .mapForDraw({{0, TriangleIdBuffer}});
+
+      // triangleFrameBuffer_.clearColor(1, Mn::Vector4i{27, 27, 27, 27});
+
+      CORRADE_INTERNAL_ASSERT(
+          framebuffer_.checkStatus(Mn::GL::FramebufferTarget::Draw) ==
+          Mn::GL::Framebuffer::Status::Complete);
+
+      triangleMesh_ = Mn::GL::Mesh{};
+      triangleMesh_.setCount(3);
```

Comment:
This is just going to segfault (or fail silently) as there isn't a vertex buffer (or index buffer) getting uploaded to the GPU.

This really shouldn't be done in the RenderTarget (the depth unprojection shader is a special beast -- it is a "compute shader" -- which is why it's in the render target).  I believe this can be done with the ShaderManager from https://github.com/facebookresearch/habitat-sim/pull/489 and just draw to the existing ObjectID attachment on the main frame buffer (cc @bigbike @matthewjmay)?

Reply:
@aclegg3 : What do you think?

Reply:
> This is just going to segfault (or fail silently) as there isn't a vertex buffer (or index buffer) getting uploaded to the GPU.

Just FYI, attribute-less index-less meshes are fine :) That's how a [fullscreen triangle](https://doc.magnum.graphics/magnum/namespaceMagnum_1_1MeshTools.html#aba8238805e0e572ea2edb8b2bca8c35c) is usually implemented, with an input-less shader that generates vertices from `gl_VertexID`.

Reply:
Ooo fancy -- I should stop doubting that magnum won't make OpenGL just do the right thing.

Reply:
I updated the PR by modifying the vert and frag shaders, creating a TriangleIDDrawable class, and adding it to ResourceManager.cpp. What I don't understand is the connection between the new shader I wrote, the drawable, and TriangleIdBuffer inside Rendertarget.cpp.